cdap-docs/reference-manual/source/release-notes.rst

```diff
@@ -28,26 +28,280 @@ Cask Data Application Platform Release Notes
+  ``batchaggregator`` plugin type. An aggregator operates on a collection of records,
+  grouping them by a key and performing an aggregation on each group.
+
+- `CDAP-5172 <https://issues.cask.co/browse/CDAP-5172>`__ - Added support for
+  authorization extensions in CDAP. Extensions extend an ``Authorizer`` class and provide a
+  bundle jar containing all their required dependencies. This jar is then specified using
+  the property ``security.authorization.extension.jar.path`` in the ``cdap-site.xml``.
+
+- `CDAP-5191 <https://issues.cask.co/browse/CDAP-5191>`__ - Added an ``FTPBatchSource``
+  that can fetch data from an FTP server in a batch pipeline of Cask Hydrator.
+
+- `CDAP-5205 <https://issues.cask.co/browse/CDAP-5205>`__ - Added a global search across
+  all CDAP entities in the CDAP UI.
+
+- `CDAP-5274 <https://issues.cask.co/browse/CDAP-5274>`__ - The Cask Hydrator Studio now
+  includes the capability to configure a new type of pipeline, a "data pipeline" (beta
+  feature).
+
+- `CDAP-5360 <https://issues.cask.co/browse/CDAP-5360>`__ - The CDAP UI now supports
+  ``Sparksink`` and ``Sparkcompute`` plugin types, included in a new "data pipeline"
+  artifact.
+
+- `CDAP-5361 <https://issues.cask.co/browse/CDAP-5361>`__ - Added a ``SparkTransform``
+  plugin type, which allows the running of a Spark job that operates as a transform in an ETL
+  batch pipeline.
+
+- `CDAP-5362 <https://issues.cask.co/browse/CDAP-5362>`__ - Added a ``SparkSink`` plugin
+  type, which allows the running of a Spark job (such as machine learning) on the output of
+  an ETL batch pipeline.
+
+- `CDAP-5392 <https://issues.cask.co/browse/CDAP-5392>`__ - Added support for
+  ``FormatSpecification`` in Spark when consuming data from a stream.
 
 Improvements
 ------------
-- `CDAP-9999 <https://issues.cask.co/browse/CDAP-9999>`__ - Added...
+- `CDAP-4071 <https://issues.cask.co/browse/CDAP-4071>`__ - MapReduce programs can now be
+  configured to write metadata for each partition created using a ``DynamicPartitioner``.
+
+- `CDAP-4117 <https://issues.cask.co/browse/CDAP-4117>`__ - Fixed an issue of not using
+  the correct user account to access HDFS when submitting a YARN application through
+  Apache Twill, which caused a cleanup failure (and a confusing error message) upon
+  application termination.
+
+- `CDAP-4644 <https://issues.cask.co/browse/CDAP-4644>`__ - Workflow logs now contain logs
+  from all of the actions executed by a workflow.
+
+- `CDAP-4842 <https://issues.cask.co/browse/CDAP-4842>`__ - Added a ``hydrator-test``
+  module that contains mock plugins for unit testing Hydrator plugins.
+
+- `CDAP-4925 <https://issues.cask.co/browse/CDAP-4925>`__ - Added to the CDAP test
+  framework the ability to delete applications and artifacts, retrieve application
+  information, update an application, and write and remove properties for artifacts.
+
+- `CDAP-4955 <https://issues.cask.co/browse/CDAP-4955>`__ - Added a 'postaction' Cask
```

Comment:
@albertshau Are plugin types also part of "Cask Hydrator"? In other words, is Hydrator just the plugin implementations?